### PR TITLE
Chat service backend and Chatbox component

### PR DESCRIPTION
**Pull request by hiongkaihan**
Chat service backend and Chatbox component added

Ignore changes in App.js and Chat.js

### CODE REVIEW

**Comment by TheSpaceCuber** [comment]
Will review by Wed night.

File: chat-service/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+const io = new Server(server, {
+    cors: {
+        origin: '*'
+    },
+})
+
+const CHAT_BOT = 'ChatBot'; // Constants and Magic Numbers will be abstracted when finalising
+io.on('connection', (socket) => {
+    console.log(`User connected ${socket.id}`)
```
Comment (TheSpaceCuber) [inline chat-service/index.js:22]: Logging: User connected to socket `${socket.id}` may be clearer instead

File: frontend/src/components/Chat.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useEffect } from "react";
+import io from 'socket.io-client';
+import ChatBox1 from "./ChatBox";
+
+
+// Ignore this component
+// This component is merely for testing purposes, will be removed when integrated into InterviewPage.js
+const Chat = ({user}) => {
+    const socket = io.connect('http://localhost:8005'); // our server will run on port 8005, so we connect to it from here
```
Comment (TheSpaceCuber) [inline frontend/src/components/Chat.js:9]: Can put this URL as a variable instead?

Comment (TheSpaceCuber) [reply frontend/src/components/Chat.js:9]: Ok but if it is going to be removed then don't need.

File: frontend/src/components/ChatBox.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import { useState, useEffect, useRef } from 'react'
+import { Container , Card, InputGroup, Button, Form } from 'react-bootstrap'
+
+const ChatBox = ({ socket, username, room }) => {
+    const [messagesReceived, setMessagesReceived] = useState([])
+    const [message, setMessage] = useState('')
+    const chatBody = useRef(null)
+    const chatInput = useRef(null)
```
Comment (TheSpaceCuber) [inline frontend/src/components/ChatBox.js:8]: What is the useRef hook used for?

**Review by TheSpaceCuber — COMMENTED** [review]
Preliminary comments

File: frontend/src/components/ChatBox.js (hunk 3 above)
Comment (hiongkaihan) [reply frontend/src/components/ChatBox.js:8]: Allows me to access DOM elements by individually selecting/referencing them

image
image
image

File: chat-service/index.js (hunk 1 above)
Comment (hiongkaihan) [reply chat-service/index.js:22]: Fixed

File: frontend/src/components/ChatBox.js (hunk 3 above)
Comment (TheSpaceCuber) [reply frontend/src/components/ChatBox.js]: Ok

**Review by TheSpaceCuber — APPROVED** [review]
LGTM, tested without issues faced.

**Comment by TheSpaceCuber** [comment]
Good for @Herrekt to read Interview.js first before merging.

**Pull request merged**